lib/formalism/form/validation_error.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Formalism
+	class Form < Action
+		## When trying to get `#result` from unsuccessful outcome
+		class ValidationError < StandardError
+			def initialize(errors)
+				super(
+					"Outcome has errors: #{errors.to_a}"
+				)
+			end
```

Comment:
Maybe this will be more usefull

```ruby
attr_reader :errors

def initialize(errors)
  @errors = errors 
  super("Outcome has errors: #{errors.to_a}")
end
```

Reply:
Also why not `errors.inspect`?

Reply:
> Maybe this will be more usefull

Maybe. But it can be added later.

> Also why not errors.inspect?

`errors` is instance of `Set`.

Reply:
> Maybe. But it can be added later.

I think if it is acceptable it should be added as soon as possible, because it makes public API more consistent. Otherwise if it makes public API worse (I.e. you don't want people to handle this error) it should not be added in future.

I find it usefull to handle validation errors like this in controller / middleware

> errors is instance of Set.

Set has nice inspect, `"#<Set: {1, 2, 3}>"`. And also errors not really forced to be a set, it can be overriden

Reply:
> I think if it is acceptable it should be added as soon as possible, because it makes public API more consistent.

I'm not sure that somebody needs for this. And adding a feature isn't breaking change, so it's OK.

> I.e. you don't want people to handle this error

Why? People can just catch this error and do their things. 

It's exception, for exceptional case. When you're calling `#result` for unsuccessful `outcome` — you can get `errors` from this `outcome`.

> I find it usefull to handle validation errors like this in controller / middleware

I think condition (`outcome.success?`) is better, then rescuing exception.

> Set has nice inspect

I don't agree.

> And also errors not really forced to be a set, it can be overriden

Yes. But, I think, it'll be `Enumerable` and have `#to_a` method.

Reply:
Done, thank you.